### ISSUE TITLE
getRealtimeLogs not working

### ISSUE THREAD

**Issue by dev-iot-jogja** [caobo171/node-zklib#26]
Here my code

```
const ZKLib = require('./zklib')

let machine_ip = '192.168.192.168';

const test = async () => {
    let zkInstance = new ZKLib(machine_ip, 4370, 10000, 4000);
    try {
        await zkInstance.createSocket()
        console.log(await zkInstance.getInfo())
    } catch (e) {
        console.log("init", e);
    }

    try{
        zkInstance.getRealTimeLogs((data)=>{
            // do something when some checkin 
            console.log('real time', data)
        })
    } catch (e) {
        console.log("realtime error", e)
    }
}

test()
```

when I try to log, I found that checkNotEventTCP on getRealTimeLogs return CMD_ACK_OK not EF_ATTLOG or CMD_REG_EVENT

what should I do to make getRealTimeLogs working?

**Comment by mahrous78** [comment]
getRealTimeLogs is not working with me too

**Comment by caobo171** [comment]
Sorry for responding lately, but did  you guys make sure that there'is no another machine connect to that device. Because even if it connected successfully but realtime event is not allowed for multiple devices

**Comment by herormo** [comment]
Not working for me too, there is only one machine connected to the device. @dev-iot-jogja did you manage to fix this?

**Comment by danishthetechie** [comment]
No it is not connected. Could you please have a look, I tried several times.. I really need to do this pal. We can even skype, or call

**Comment by danishthetechie** [comment]
No other machine is connected. but still not working

**Comment by MoradAbdelgaber** [comment]
@caobo171  any news here for this bug . still getting nothing with getRealTimeLogs

**Comment by AlifElahi** [comment]
Hey, I am also facing the same issue. Has anyone found any workaround? (@danishthetechie, @MoradAbdelgaber )
 @caobo171 brother have you happened to have some time to look at this?

**Comment by ebgutierrez** [comment]
Hi, I'm having the same issue here. Is there any update on this?

**Comment by danishthetechie** [comment]
No, haven't found any resolution regarding this. still my work is pending on getRealTimeLogs issue

**Comment by mahrous78** [comment]
the python lib works fine

**Comment by JorgeSolisC** [comment]
@danishthetechie encontraste alguna solución?

**Comment by JorgeSolisC** [comment]
@dev-iot-jogja did you find a solution?

**Comment by danishthetechie** [comment]
> @dev-iot-jogja did you find a solution?

No,
no solution found for real time logs

**Comment by aadrsh** [comment]
> when I try to log, I found that checkNotEventTCP on getRealTimeLogs return CMD_ACK_OK not EF_ATTLOG or CMD_REG_EVENT

@caobo171 @dev-iot-jogja 
Yeah same here, Found that getRealTimeLogs function is exiting after checkNotEventTCP check
so I tried commenting out the check
I found out that the packet returned by machine is Regular Packet instead of a Realtime Packet, 

Machine Reply after requesting Realtime data is still a Regular Packet
<Buffer 50 50 82 7d 08 00 00 00 d0 07 45 e5 e7 12 03 00> 

Referring to https://github.com/adrobinoga/zk-protocol/blob/master/protocol.md
First 2 bytes of Regular Packet is Command ID, next 2 Bytes are Checksum, next 2 are sessionId and rest are reply  number and data

But in case of realtime packet CommandID should be 500 But it is ACK_OK i.e. 2000 and instead of realtime event id it sends session ID thats why checkEventNotTCP is true

So.. what I think is the machine isn't sending realtime packets or if its sending then we are not receiving it
I am not good with sockets or js but this is what i came up with

**Comment by shahria7k** [comment]
> the python lib works fine
which lib?

**Comment by daoues1** [comment]
> the python lib works fine
@mahrous78 it is'nt working for me with the python lib could you explain how did u make it work ?

**Comment by daoues1** [comment]
Nope still

**Comment by catajoynick** [comment]
Oh, then I cant use this library then. Thanks for reply.

**Comment by chermakani** [comment]
HI, I am also facing the same issue.Anyone found solution?@caobo171  can you please look at this issue?Let us know if you found solution

**Comment by zhandos38** [comment]
I am also facing the same issue and so tired please help

**Comment by justlester** [comment]
try adding await zkInstance.enableDevice(); before  zkInstance.getRealTimeLogs
